apps/internal/oauth/oauth.go

```diff
@@ -194,14 +195,31 @@ func (d DeviceCode) Token(ctx context.Context) (accesstokens.TokenResponse, erro
 	}
 }
 
-var waitRE = regexp.MustCompile("(authorization_pending|slow_down)")
+type deviceCodeError struct {
+	Error string `json:"error"`
+}
 
-// TODO(msal): This is freaking terrible. The original just looked for the exact word in the error output.
-// I doubt this worked. I don't know if the service really does this, but it should send back a structured
-// error response. Anyways, I updated this to search the entire return error message, which will be the body
-// of the return.
 func isWaitDeviceCodeErr(err error) bool {
-	return waitRE.MatchString(err.Error())
+	var c errors.CallErr
+	if errors.As(err, &c) {
```

Comment:
Nit: I'd flip this condition to remove the nesting.

Reply:
Resolved in this commit : https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/178/commits/da3abaa86143db33a44497c61f65f67cc79e6872